src/pages/Filaments.tsx

```diff
@@ -123,6 +130,14 @@ const Filaments = () => {
           </table>
         </div>
       )}
+      <ConfirmationDialog
+        isOpen={isDialogOpen}
+        setIsOpen={setIsDialogOpen}
+        name={deletingFilament?.name}
+        deleteFn={() => {
+          if (deletingFilament) remove(deletingFilament);
+        }}
+      />
```

Comment:
A little bit of performance optimization. If there's no `deletingFilament`, the `ConfirmationDialog` component won't be rendered, and we gain a little bit of code improvement without the `if` statement and the optional chaining operator.


```suggestion
      {deletingFilament && (
        <ConfirmationDialog
          isOpen={isDialogOpen}
          setIsOpen={setIsDialogOpen}
          name={deletingFilament.name}
          deleteFn={() => remove(deletingFilament)}
        />
      )}
```